xmppserver/src/main/java/org/jivesoftware/openfire/OfflineMessageStore.java

```diff
@@ -79,11 +83,23 @@
         "DELETE FROM ofOffline WHERE username=?";
     private static final String DELETE_OFFLINE_MESSAGE =
         "DELETE FROM ofOffline WHERE username=? AND creationDate=?";
+    private static final String DELETE_OFFLINE_MESSAGE_BEFORE =
+            "DELETE FROM ofOffline creationDate < ?";
+    private static final String SELECT_SIZE_OFFLINE_ALL_USERS =
+            "SELECT SUM(messageSize),username FROM ofOffline group by username";
 
     private static final int POOL_SIZE = 10;
-    
+
     private Cache<String, Integer> sizeCache;
 
+    /**
+     * Members for automatic offline message cleaning
+     * */
+    private int daystolive = 14; //days
```

Comment:
A default of 14 days is way to short for my feeling. Lets err on the side of caution, and use a year or something.

Reply:
increased it to 365